netsim/modules/gateway.py

```diff
@@ -153,8 +153,50 @@ def module_init(self, topology: Box) -> None:
         process_gw_id(vdata.get('gateway.id',None),required=False,path=f'VLAN {vname}')
 
   def link_pre_link_transform(self, link: Box, topology: Box) -> None:
-    if not 'gateway' in link:
-      return
+    #
+    # Step#1: deal with 'gateway' interface attributes
+    #
+    # If an interface has 'gateway' parameter (and it's not False), we must enable
+    # gateway on the link and remember that we have to turn 'gateway' off on other
+    # interfaces if needed
+    #
+    # Also: combining 'gateway:False' on link with 'gateway:True' on interface is
+    # a stupidity and will be rejected
+    #
+    copy_to_link = False
+    for intf in link.interfaces:
+      if 'gateway' not in intf:                             # No 'extra special' stuff, move on
+        continue
+      if intf.get('gateway',None) is False:                 # The 'special' stuff is don't do it?
+        continue                                            # ... sure, I can do that ;)
+      if 'gateway' in link:                                 # Do we already have 'gateway' on link?
+        if link.gateway is False:                           # If we have 'gateway:false' we're dealing with a stupidity
+          log.error(
+            'Cannot combine gateway:false on link with gateway:true on interface',
```

Comment:
```gateway``` on interface could also be a dict setting the vrrp.priority for example

"Cannot combine gateway:false on link with gateway attribute on interface" (although I could come up with scenarios where that could make sense to me, I'll spare you the eye-rolling exercise this time)

Reply:
```Warning: interface gateway attributes ignored due to link.gateway=False```